c7n/resources/sagemaker.py

```diff
@@ -0,0 +1,307 @@
+# Copyright 2016-2017 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from botocore.exceptions import ClientError
+
+from c7n.manager import resources
+from c7n.filters import FilterRegistry
+from c7n.query import QueryResourceManager
+from c7n.utils import local_session, type_schema
+from c7n.actions import BaseAction
+from c7n.tags import RemoveTag, Tag, TagActionFilter, TagDelayedAction
+
+
+filters = FilterRegistry('sagemaker.filters')
+filters.register('marked-for-op', TagActionFilter)
+
+
+@resources.register('notebook-instance')
```

Comment:
notebook feels a little ambigious in a flat namespace of resource types, how about sagemaker-instance instead.

Reply:
actually how about `sagemaker-notebook`

Reply:
that sounds good... also helps clarify the resource... notebook-instance is a little vague